src/CustomList.php

```diff
@@ -85,6 +85,34 @@ public function offsetUnset(mixed $offset): void
         unset($this->items[$offset]);
     }
 
+    /**
+     * Add list to list
+     *
+     * @param CustomList $list
+     * @return void
+     * @throws CustomListTypeErrorException
+     */
+    public function concatenate(CustomList $list): void
+    {
+        $this->items = array_merge($this->items, $list->items);
+    }
```

Comment:
You could evolve this function using the spread operator to allow to concatenate one or more CustomLists to the current list:

```suggestion
     /**
     * Concatenate to the current list the items from one or more lists
     *
     * @param mixed ...$lists
     * @return void
     * @throws CustomListTypeErrorException
     */
    public function concatenate(...$lists): void
    {
        foreach ($lists as $list) {
            $this->items = array_merge($this->items, $list->get());
        }

        $this->checkItems();
    }
```

what do you think?